### PR TITLE
remove endpoint validation

### PR DESCRIPTION
**Pull request by xinlian12**
## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
Bug fix.
Customer is getting "Invalid endpoint: Invalid host: xxx" exception even though the endpoint is valid.

### CODE REVIEW

**Comment by ryancrawcour** [comment]
i believe this was added by the Confluent team for validation when entering config in their managed cloud. I am ok with removing it if local is a valid value for endpoint.

**Review by ryancrawcour — APPROVED** [review]
i believe this was added by the Confluent team for validation when entering config in their managed cloud. I am ok with removing it if local is a valid value for endpoint.

**Review by kirankumarkolli — APPROVED** [review]

**Review by FabianMeiswinkel — DISMISSED** [review]
Thanks! LGTM - I would just add one unit test for regression converage in src/test/java/com/azure/cosmos/kafka/connect/CosmosConfigTest.java

**Comment by xinlian12** [comment]
> i believe this was added by the Confluent team for validation when entering config in their managed cloud. I am ok with removing it if local is a valid value for endpoint.

> Thanks! LGTM - I would just add one unit test for regression converage in src/test/java/com/azure/cosmos/kafka/connect/CosmosConfigTest.java

Thanks, added

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java [hunk 1]
```diff
@@ -211,10 +211,6 @@ public static void validateEndpoint(String endpoint) throws URISyntaxException,
         if (!VALID_ENDPOINT_COSMOS_INSTANCE_PATTERN.matcher(cosmosInstance).matches()) {
```
Comment (milismsft) [inline src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:211]: do we need this thing as well? what's the pattern that we expect? What if the user specifies and IP address (ie see the local Cosmos DB emulator)?

**Comment by xinlian12** [comment]
@fzhong-connect @ryancrawcour can u share more context about the endpoint validation?
 why we need the specific check for the endpoint? 
 If it is an invalid endpoint, the underlying SDK will throw the exception. Why not just relying on it?
 And if I remove the endpoint validation, what impact will have for confluent cloud platform?

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java (hunk 1 above)
Comment (ryancrawcour) [reply src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:211]: these validations are for the config UX to validate that users have entered valid values for various config settings. 
this is similar to the config checks on the portal when users supply a value for their endpoint, database name etc. 

the pattern is here - https://github.com/microsoft/kafka-connect-cosmosdb/blob/20cff56f8b03b0b51ab9df521182c8992351b080/src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java#L30

[a-z0-9\\-]{3,44}

**Comment by ryancrawcour** [comment]
> @fzhong-connect @ryancrawcour can u share more context about the endpoint validation? why we need the specific check for the endpoint? If it is an invalid endpoint, the underlying SDK will throw the exception. Why not just relying on it? And if I remove the endpoint validation, what impact will have for confluent cloud platform?
> 
> Thanks

this is merely UX config value validation. same as the azure portal does when users create accounts and supply values for things like account name, db & container names etc. 

yes, the SDK will throw errors if the values are incorrect and the connector can't connect, but these validations are to prevent (safeguard) against incorrect configuration. 

we can review which are needed or not and amend as needed. 

using the local emulator with Kafka Connect would be interesting as you'd have to run the emulator on the same environment as Kafka Connect. I mean it is theoretically possible to do, and if it is causing actual errors with customers then we should re-evaluate which validations we perform.

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
Overall looks good, missing changelog entry.

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**